substrate-archive-common/src/util.rs

```diff
@@ -96,30 +95,25 @@ fn format_opt(file: Option<String>) -> String {
 /// Linux | $XDG_DATA_HOME or $HOME/.local/share/substrate_archive | /home/alice/.local/share/substrate_archive/
 /// macOS | $HOME/Library/Application Support/substrate_archive | /Users/Alice/Library/Application Support/substrate_archive/
 /// Windows | {FOLDERID_LocalAppData}\substrate_archive | C:\Users\Alice\AppData\Local\substrate_archive
-pub fn substrate_dir() -> Result<PathBuf, ArchiveError> {
-	if let Some(base_dirs) = dirs::BaseDirs::new() {
-		let mut path = base_dirs.data_local_dir().to_path_buf();
-		path.push("substrate_archive");
-		Ok(path)
-	} else {
-		Err(ArchiveError::from("No valid home directory path could be retrieved from the operating system"))
-	}
+pub fn substrate_dir() -> io::Result<PathBuf> {
+	let base_dirs = dirs::BaseDirs::new().ok_or_else(|| {
+		io::Error::new(
+			io::ErrorKind::Other,
+			"No valid home directory path could be retrieved from the operating system",
+		)
+	})?;
+	let mut path = base_dirs.data_local_dir().to_path_buf();
+	path.push("substrate_archive");
+	Ok(path)
 }
 
 /// Create an arbitrary directory on disk.
-pub fn create_dir(path: &Path) -> Result<(), ArchiveError> {
-	if let Err(e) = std::fs::create_dir_all(path) {
-		match e.kind() {
-			std::io::ErrorKind::AlreadyExists => (),
-			_ => return Err(ArchiveError::from(format!("A directory in the path '{:?}' could not be created", path))),
+pub fn create_dir(path: &Path) -> io::Result<()> {
+	if let Err(err) = fs::create_dir_all(path) {
+		match err.kind() {
+			io::ErrorKind::AlreadyExists => (),
+			_ => return Err(err),
```

Comment:
I suspect this change will make it harder to understand the error if this fails because `io::Error` is quite tricky to understand/debug but it's fine.